src/instance/DataModelInstance.ts

```diff
@@ -13,24 +16,37 @@ import {
   buildAuxiliaryKeyMap,
   findBestCandidate
 } from './util';
-import { DerivedType, LeafRefType } from '../types';
-import DataModel, { Model, List, Leaf, Choice } from '../model';
-import { isElement } from '../util/xmlUtil';
 
 export default class DataModelInstance {
   public rawInstance: Element;
   public model: DataModel;
   public root: Map<string, ContainerInstance>;
 
-  constructor(model: DataModel, instance: Element) {
+  constructor(model: DataModel, instance: Element | { [rootName: string]: IContainerJSON }) {
     this.model = model;
+    this.root = new Map();
 
     const rootName = [...model.root.keys()][0];
 
-    this.rawInstance = instance;
-
-    this.root = new Map();
-    this.root.set(rootName, new ContainerInstance(model.root.get(rootName), instance.get('./*[1]')));
+    if (instance instanceof Element) {
+      this.rawInstance = instance;
+      this.root.set(rootName, new ContainerInstance(model.root.get(rootName), instance.get('./*[1]')));
+    } else {
+      this.root.set(rootName, new ContainerInstance(model.root.get(rootName), Object.values(instance)[0]));
```

Comment:
Trivial: can DRY this by assigning before the conditionals `model.root.get(rootName)`